src/Symfony/Component/Security/Http/Firewall/AccessListener.php

```diff
@@ -26,17 +24,13 @@
  */
 class AccessListener implements ListenerInterface
 {
-    private $tokenStorage;
-    private $accessDecisionManager;
     private $map;
-    private $authManager;
+    private $authorizationChecker;
 
-    public function __construct(TokenStorageInterface $tokenStorage, AccessDecisionManagerInterface $accessDecisionManager, AccessMapInterface $map, AuthenticationManagerInterface $authManager)
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker, AccessMapInterface $map)
```

Comment:
This is a BC break, so it can't be merged as is. Not sure if BC breaks should be added in older versions or if we should not see this class as an extension point (which it is atm).

Reply:
even then, we still need to preserve BC even in master

Reply:
sure, I'll prepare BC changes in near time.